core/model/modx/modscript.class.php

```diff
@@ -78,6 +78,12 @@ public function process($properties= null, $content= null) {
                 if ($this->_output && is_string($this->_output)) {
                     /* collect element tags in the evaluated content and process them */
                     $maxIterations= intval($this->xpdo->getOption('parser_max_iterations',null,10));
+
+                    // Make sure we have the parser available
+                    if ($this->xpdo->parser === null) {
```

Comment:
This should never happen, and if it does it would have to be something setting the parser to null explicitly. `modScript::process()` calls `parent::process()` which already explicitly loads the parser service. See https://github.com/modxcms/revolution/blob/2.x/core/model/modx/modelement.class.php#L252.